enterprise/app/workflows/workflows.tsx

```diff
@@ -415,6 +418,21 @@ class RepoItem extends React.Component<RepoItemProps, RepoItemState> {
       .executeWorkflow(
         new workflow.ExecuteWorkflowRequest({
           pushedRepoUrl: this.props.repoUrl,
+          // Parse "var1=val1,var2=val2" string from the input field to Record<string, string>
+          env: this.state.runWorkflowEnv
+            .split(",")
+            .filter((n) => n.includes("="))
+            .map((n) => n.trim().split("="))
+            .filter(([k, v]) => k && v)
+            .map(([k, v]) => [k.trim(), v.trim()])
```

Comment:
nit: handle values with `=`

```suggestion
            .filter((parts) => parts.length)
            .map(([name, ...valueParts]) => [name, valueParts.join("=")])
```

Reply:
also `Object.fromEntries()` can help you form the env map from your `[k, v]` pairs

Reply:
Nice tip. Using fromEntries saved me from that .reduce 🙌